### PR TITLE
fix: allow `new DataSource(connector, settings)`

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

One more edge case to keep it backward compatible.

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: lib/datasource.js [hunk 1]
```diff
@@ -347,11 +347,17 @@ DataSource.prototype.setup = function(dsName, settings) {
...
-    settings = dsName;
-    dsName = undefined;
+  // First argument is an `object`
+  if (dsName && typeof dsName === 'object') {
+    if (settings === undefined) {
+      // setup({name: 'myDataSource', connector: 'memory'})
+      settings = dsName;
+      dsName = undefined;
+    } else {
+      // setup(connector, {name: 'myDataSource', host: 'localhost'})
```
Comment (bajtos) [inline lib/datasource.js]: Please add this example usage to the jsdoc for `setup` and ideally to jsdoc for `new DataSource()` too.
